Alt Text and Link Edits - Pt. 4 (iOS) (#3342)
braze-inc/braze-docs: _docs/_developer_guide/platform_integration_guides/ios/advanced_use_cases/linking.md

article_title: Deep Linking for iOS
platform: iOS
page_order: 0
description: "This article covers how to implement the universal deep linking delegate for your iOS app and examples on how to deep link to app settings or a News Feed."

---

# Linking

## Deep links

> For information regarding what a deep link is, see our [FAQ Section][4]. If you're looking to implement deep links for the first time, see the documentation below.

Reviewer: ```suggestion
For more information on deep links, refer to our [FAQ section][4]. If you're looking to implement deep links for the first time, see the documentation below.
```

You can handle ATS in one of the following three ways:

#### Ensure all links are ATS-compliant (recommended)

> Your Braze integration can satisfy ATS requirements by ensuring that any existing links you drive users to (through in-app message and push campaigns or News Feed cards) satisfy ATS requirements. While there are ways to bypass ATS restrictions, Braze's recommended best practice is  to ensure that all linked URLs are ATS-compliant. Given Apple's increasing emphasis on application security, the below-listed approaches to allowing ATS exceptions are not guaranteed to be supported by Apple.

Reviewer: ```suggestion
Your Braze integration can satisfy ATS requirements by ensuring that any existing links you drive users to (through in-app message and push campaigns or News Feed cards) satisfy ATS requirements. While there are ways to bypass ATS restrictions, Braze's recommended best practice is to ensure that all linked URLs are ATS-compliant. Given Apple's increasing emphasis on application security, the following approaches to allowing ATS exceptions are not guaranteed to be supported by Apple.
```

### Default WebView customization

The open-source `ABKModalWebViewController` class displays web URLs opened by the SDK, typically when "Open Web URL Inside App" is selected for a web deep link.

> You can declare a category for, or directly modify, the `ABKModalWebViewController` class to apply customization to the web view. Check the class's [.h file][6] and [.m file][5] for more detail.

Reviewer: ```suggestion
You can declare a category for, or directly modify, the `ABKModalWebViewController` class to apply customization to the web view. Check the class' [.h file][6] and [.m file][5] for more detail.
```

### Linking handling customization

> The `ABKURLDelegate` protocol can be used to customize the handling of URLs such as deep links, web URLs, and Universal Links. To set the delegate during Braze initialization, pass a delegate object to the `ABKURLDelegateKey` in the `appboyOptions` of [`startWithApiKey:inApplication:withAppboyOptions:`][22]. Braze will then call your delegate's implementation of `handleAppboyURL:fromChannel:withExtras:` before handling any URIs.

Reviewer: ```suggestion
The `ABKURLDelegate` protocol can be used to customize the handling of URLs such as deep links, web URLs, and universal links. To set the delegate during Braze initialization, pass a delegate object to the `ABKURLDelegateKey` in the `appboyOptions` of [`startWithApiKey:inApplication:withAppboyOptions:`][22]. Braze will then call your delegate's implementation of `handleAppboyURL:fromChannel:withExtras:` before handling any URIs.
```
